src/choleskyQR.jl

```diff
@@ -0,0 +1,170 @@
+"""
+cholesky_jacobimatrix(w, P)
+
+returns the Jacobi matrix `X` associated to a quasi-matrix of polynomials
+orthogonal with respect to `w(x) w_p(x)` where `w_p(x)` is the weight of the polynomials in `P`.
+
+The resulting polynomials are orthonormal on the same domain as `P`. The supplied `P` must be normalized. Accepted inputs are `w` as a function or `W` as an infinite matrix representing multiplication with the function `w` on the basis `P`.
+
+An optional bool can be supplied, i.e. `cholesky_jacobimatrix(sqrtw, P, false)` to disable checks of symmetry for the weight multiplication matrix and orthonormality for the basis (use with caution).
+"""
+function cholesky_jacobimatrix(w::Function, P::OrthogonalPolynomial, checks::Bool = true)
+    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    W = Symmetric(P \ (w.(axes(P,1)) .* P)) # Compute weight multiplication via Clenshaw
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+function cholesky_jacobimatrix(W::AbstractMatrix, P::OrthogonalPolynomial, checks::Bool = true)
+    checks && !(P isa Normalized) && error("Polynomials must be orthonormal.")
+    checks && !(W isa Symmetric) && error("Weight modification matrix must be symmetric.")
+    bands = CholeskyJacobiBands(W, P)
+    return SymTridiagonal(bands[1,:],bands[2,:])
+end
+
+# The generated Jacobi operators are symmetric tridiagonal, so we store their data as two bands
+mutable struct CholeskyJacobiBands{T} <: AbstractCachedMatrix{T}
+    data::Matrix{T}         # store so-far computed block of entries 
```

Comment:
Storage should probably be two vectors corresponding to the diagonals, not a matrix.

Reply:
It is being stored as a (2,N) matrix storing the two vectors. Is there a meaningful difference between that and storing it as two separate vectors? The nice thing about this is that I only have to resize one object but I can change it if you think this matters.

Reply:
you can't `resize!` a matrix but you can two vectors. Plus `SymTridiagonal` takes vectors not matrices.